drivers/i2c/i2c_ll_stm32.c

```diff
@@ -44,6 +46,20 @@ LOG_MODULE_REGISTER(i2c_ll_stm32);
 #define STM32_I2C_DOMAIN_CLOCK_SUPPORT 0
 #endif
 
+#ifdef CONFIG_PM
+#define PM_POLICY_STATE_LOCK_ACTION(node_id, prop, idx, pm_func) \
+	pm_func( \
+		UTIL_CAT(PM_STATE_, \
+			DT_STRING_UPPER_TOKEN( \
+				DT_PHANDLE_BY_IDX(node_id, prop, idx), \
+				power_state_name \
+			) \
+		), \
+		DT_PROP_BY_PHANDLE_IDX(node_id, prop, idx, \
+			substate_id) \
```

Comment:
#### Note to self

`substate-id` is not a required property, so we should first check if it is set before generating the `pm_func` call here.